src/Datadog.Trace/Propagation/ContextPropagatorBuilder.cs

```diff
@@ -15,14 +16,19 @@ internal static class ContextPropagatorBuilder
                 { PropagatorType.Default, (convention) => new DDSpanContextPropagator(convention) },
             };
 
-        public static IPropagator BuildPropagator(PropagatorType propagator, ITraceIdConvention traceIdConvention)
+        public static ICollection<IPropagator> BuildPropagators(IEnumerable<PropagatorType> propagatorTypes, ITraceIdConvention traceIdConvention)
         {
-            if (_propagatorSelector.TryGetValue(propagator, out Func<ITraceIdConvention, IPropagator> getter))
+            return propagatorTypes.Select(type => BuildPropagator(type, traceIdConvention)).ToList();
+        }
+
+        public static IPropagator BuildPropagator(PropagatorType propagatorType, ITraceIdConvention traceIdConvention)
```

Comment:
Should it not be changed to `private`?

Reply:
I would keep it as public api. In future some vendors might choose not to use composite propagation as default.

Reply:
Just a note: it is easy to change private to public at any time. It is almost impossible to change from public to private, because somebody may already use it. So YAGNI, unless actually required, make it private.

Reply:
seems private it is